app/photo.py

```diff
@@ -40,6 +47,24 @@ class SizeTooLargeError(PhotoError):
                 'Please upload a smaller one.')
 
 
+def create_photo_with_url(handler, photo_upload, photo_url):
```

Comment:
The design might be more general-purpose to have two methods, or one method with only the handler param and an 'image' param and do some kind of type test to determine which flow to follow.

Python's weird. Parametric polymorphism would be nice.

Also, perhaps a pydoc explaining what hander is, because it wasn't clear to em.

Reply:
Added pydoc.
The reason for having two separate parameters is that all the callers will have variables for an uploaded photo and a URL, but won't know which (if either) is actually populated. I'd prefer not to have a bunch of duplicated code checking that stuff and calling the appropriate method (I'd already need two sets of it in create.py alone, for the note and record photos). So I think it does make sense to have that logic in a function, but maybe it doesn't belong in `photo.py`. What if I moved that part into `create.py`, so that `photo.py` was just about photos and `create.py` was responsible for sorting out user input and sending it to the right `photo.py` function?

Reply:
Yea, that makes sense. A little helper in `create.py` and the generic functions in `photos.py` makes sense to me.

Reply:
Done. The helper in create.py is outside the class so it can be used by other things that need to create stuff later -- the various places that create things should probably share a bunch of code, but that's a refactor for another day, so for now I put it in create.py with a todo.